operator/flags.go

```diff
@@ -234,6 +234,9 @@ func init() {
 	flags.String(option.K8sAPIServer, "", "Kubernetes API server URL")
```

Comment:
what do think about marking this one deprecated in favor of new flags ? Also a short description in this docs https://docs.cilium.io/en/stable/operations/upgrade/#admin-upgrade will be nice.

Reply:
Yeah, that makes sense to me.
The flag `--k8s-api-server` is not exposed as an helm option anyways so we can easily deprecate it in favor of the new flag.
Will update the flag and the docs.